lhs.js

```diff
@@ -42,6 +47,28 @@ class Lhs {
         }
     }
 
+    when(pred) {
+        this._predicates.push(pred);
+        return this;
+    }
+
+    /*
+     * Of all the dom nodes selected by type() or dom(), return only
+     * the fnodes that satisfy all the predicates imposed by calls to
+     * when()
+     */
+    fnodesSatisfyingWhens(fnodes) {
+        const ret = [];
+        const predicates = this._predicates;
```

Comment:
Why the introduction of this new `predicates` symbol?

Reply:
I think the forEach switches the context, so it can't access this._predicates.

Reply:
Ah, you're right. It changes `this` to point to your anonymous function. In those cases, it's idiomatic to say something like `const self = this` outside the function to preserve the old `this`. Your choice is actually superior, because it doesn't preserve anything you don't need.